server/server.js

```diff
@@ -245,13 +282,19 @@ Server.prototype._handleResourceMessage = function(socket, message, messageType)
       (this.subs[to] ? 'is subscribed' : 'not subscribed')
     );
 
-    this._persistClientData(socket, message);
-    this._storeResource(resource);
-    this._persistenceSubscribe(resource.to, socket.id);
-    this._updateLimits(socket, message, resource.options);
-    this._stampMessage(socket, message);
-    resource.handleMessage(socket, message);
-    this.emit(message.op, socket, message);
+    this._runMiddleware('post', socket, message, messageType, function (err) {
```

Comment:
ok, here's a suggestion: call them "message" filter and "resource" filter maybe. Both are essentially "before/pre" filters. All message filters get called upon receipt of the message (maybe in future, after validation). All resource filters get called after message -> resource has been done. Please also add documentation to reflect these changes.